e2e_playwright/st_vega_lite_chart_test.py

```diff
@@ -129,3 +129,27 @@ def test_empty_vega_lite_chart(app: Page, assert_snapshot: ImageCompareFunction)
 def test_check_top_level_class(app: Page):
     """Check that the top level class is correctly set."""
     check_top_level_class(app, "stVegaLiteChart")
+
+
+def test_vega_lite_chart_updates_with_slightly_different_data(
+    themed_app: Page, assert_snapshot: ImageCompareFunction
```

Comment:
nit: you can probably use `app` instead of `themed_app` to lower the number of snapshots (since the behavior doesn't really depend on theming).